### PR TITLE
Vdb 997 vat init

### PR DESCRIPTION
**Pull request by yaoandrew**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,6 @@ require (
 	github.com/pkg/errors v0.8.1 // indirect
 	github.com/sirupsen/logrus v1.4.2
 	github.com/spf13/afero v1.2.2 // indirect
-	github.com/spf13/cobra v0.0.5 // indirect
```
Comment (elizabethengelman) [inline go.mod]: I wonder if we should check this change in? 🤔

File: transformers/component_tests/triggers/vat_init_trigger_test.go [hunk 2]
```diff
@@ -83,8 +82,8 @@ var _ = Describe("Updating historical_ilk_state table", func() {
 		Expect(setupErr).NotTo(HaveOccurred())
 		expectedTimeCreated := sql.NullString{Valid: true, String: FormatTimestamp(rawTimestampTwo)}
 
-		err := repo.Create([]shared.InsertionModel{vatInitModel})
-		Expect(err).NotTo(HaveOccurred())
+		vatInitErr := event.PersistModels([]event.InsertionModel{vatInitModel}, database)
```
Comment (elizabethengelman) [inline transformers/component_tests/triggers/vat_init_trigger_test.go]: 💯

File: transformers/events/vat_init/converter.go [hunk 3]
```diff
@@ -19,38 +19,42 @@ package vat_init
...
 	"github.com/makerdao/vulcanizedb/pkg/core"
+	"github.com/makerdao/vulcanizedb/pkg/datastore/postgres"
 )
 
-type VatInitConverter struct{}
+type Converter struct{}
 
 const (
 	logDataRequired   = false
 	numTopicsRequired = 2
```
Comment (elizabethengelman) [inline transformers/events/vat_init/converter.go]: We can use the [shared converter constants ](https://github.com/makerdao/vdb-mcd-transformers/blob/staging/transformers/shared/converter.go#L32) for `logDataRequired` and then perhaps we can add a new constants for `TwoTopicsRequired`.

File: go.mod (hunk 1 above)
Comment (yaoandrew) [reply go.mod]: Probably not. I'm still puzzled sometimes by changes in my go.mod files locally.

File: transformers/component_tests/triggers/vat_init_trigger_test.go [hunk 4]
```diff
@@ -124,9 +123,13 @@ var _ = Describe("Updating historical_ilk_state table", func() {
...
-func createVatInitModel(headerId, logId int64, ilkHex string) shared.InsertionModel {
+func createVatInitModel(headerId, logId int64, ilkHex string, database *postgres.DB) event.InsertionModel {
+
+	ilkID, ilkErr := shared.GetOrCreateIlk(ilkHex, database)
+	Expect(ilkErr).NotTo(HaveOccurred())
+
 	vatInit := test_data.VatInitModel
-	vatInit.ForeignKeyValues[constants.IlkFK] = ilkHex
+	vatInit.ColumnValues[constants.IlkColumn] = ilkID
 	vatInit.ColumnValues[constants.HeaderFK] = headerId
```
Comment (m0ar) [inline transformers/component_tests/triggers/vat_init_trigger_test.go]: Use `event` package constants for non-MCD variables (`headerFK`, `logFK`, `addressFK`) :) 

We'll remove these from the vdb-mcd-transformers repo later

File: transformers/events/vat_init/converter.go [hunk 5]
```diff
@@ -19,38 +19,37 @@ package vat_init
...
-			ColumnValues: shared.ColumnValues{
-				constants.HeaderFK: log.HeaderID,
-				constants.LogFK:    log.ID,
+			OrderedColumns: []event.ColumnName{
+				event.HeaderFK, constants.IlkColumn, event.LogFK,
 			},
-			ForeignKeyValues: shared.ForeignKeyValues{
-				constants.IlkFK: ilk,
+			ColumnValues: event.ColumnValues{
+				event.HeaderFK:      log.HeaderID,
```
Comment (m0ar) [inline transformers/events/vat_init/converter.go]: :ok_hand:  `event` :ok_hand:

**Pull request merged**